src/main/java/bobby/ui/Bobby.java

```diff
@@ -0,0 +1,366 @@
+package bobby.ui;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Bobby is a chatbot that can manage a list of tasks, ToDos, Deadlines, and Events.
+ * It can mark tasks as done, unmark, delete and add tasks.
+ */
+public class Bobby {
+
+    /**
+     * Greeting message displayed when the bot starts
+     */
+    private static final String GREETING = "Hello! I'm Bobby\n"
+            + "What can I do for you?\n";
+
+    /**
+     * Exit msg when the bot terminates
+     */
+    private static final String EXIT = "Bye. Hope to see you again soon!";
+
+    /**
+     * Scanner to read user input
+     */
+    private static Scanner scan = new Scanner(System.in);
+
+    /**
+     * Storage instance for saving/loading tasks from a file
+     */
+    private static Storage storage;
+
+    /**
+     * TaskList instance to manage the list of task
+     */
+    private static TaskList taskList;
+
+    /**
+     * Parser instance to handle command parsing
+     */
+    private static Parser parser;
+
+    /**
+     * File path to store task data
+     */
+    private static final String FILE_PATH = "/Users/zhiyi/Desktop/CS2103T/Chatbot/src/main/task.txt";
+
+    /**
+     * Enum representing the basic types of tasks that Bobby can handle.
+     * T - ToDo
+     * D - Deadline
+     * E - Event
+     */
+    private enum SimpleType {
+        T, D, E
+    }
+
+    /**
+     * Enum representation of the action that Bobby can take.
+     */
+    public enum ActionType {
+        bye, list, delete, mark, unmark, todo, deadline, event, retry, find
+    }
+
+    /**
+     * Constructs a Task object from a string description.
+     *
+     * @param desc The string description of the task.
+     * @return A Task object corresponding to the description,
+     *      or null if the input is invalid
+     */
+
+    public static Task constructTask(String desc) {
```

Comment:
Well-organised code! Perhaps you could further encapsulate this function into another class i.e. Parser that deals with making sense of the users commands?